[INFO -- 2018-12-08 14:16:52]
Starting flip_value script
version: 1.1
[INFO -- 2018-12-08 14:16:52]
Initialization strings:
[INFO -- 2018-12-08 14:16:52]
(1) file /tmp/knapsack/sa
[INFO -- 2018-12-08 14:16:52]
initSignal:0
[INFO -- 2018-12-08 14:16:52]
endSignal:9
[INFO -- 2018-12-08 14:16:52]
faultModel:2
[EXCEPTION -- 2018-12-08 14:16:54]
pythonException: Unable to exit loop in struct fields; Exiting wihtout making a bit flip
[EXCEPTION -- 2018-12-08 14:16:55]
pythonException: empty range for randrange() (0,0, 0)
[INFO -- 2018-12-08 14:16:55]
Backtrace BEGIN:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff6ede801 in __GI_abort () at abort.c:79
#2  0x00007ffff6f27897 in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7054b9a "%s\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007ffff6f2e90a in malloc_printerr (str=str@entry=0x7ffff70567a8 "munmap_chunk(): invalid pointer") at malloc.c:5350
#4  0x00007ffff6f35ecc in munmap_chunk (p=0x5555557720a5) at malloc.c:2846
#5  __GI___libc_free (mem=0x5555557720b5) at malloc.c:3117
#6  0x0000555555556d66 in __gnu_cxx::new_allocator<int>::deallocate (this=0x5555557590a0 <wt2>, __p=0x5555557720b5) at /usr/include/c++/7/ext/new_allocator.h:125
#7  0x0000555555556912 in std::allocator_traits<std::allocator<int> >::deallocate (__a=..., __p=0x5555557720b5, __n=242) at /usr/include/c++/7/bits/alloc_traits.h:462
#8  0x0000555555556176 in std::_Vector_base<int, std::allocator<int> >::_M_deallocate (this=0x5555557590a0 <wt2>, __p=0x5555557720b5, __n=242) at /usr/include/c++/7/bits/stl_vector.h:180
#9  0x0000555555555eb1 in std::_Vector_base<int, std::allocator<int> >::~_Vector_base (this=0x5555557590a0 <wt2>, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:162
#10 0x0000555555555949 in std::vector<int, std::allocator<int> >::~vector (this=0x5555557590a0 <wt2>, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:435
#11 0x00007ffff6ee1041 in __run_exit_handlers (status=0, listp=0x7ffff7289718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at exit.c:108
#12 0x00007ffff6ee113a in __GI_exit (status=<optimized out>) at exit.c:139
#13 0x000055555555504b in exit_gracefully () at knapsacksa.cpp:40
#14 0x0000555555555200 in cost (solution=std::vector of length 200, capacity 200 = {...}) at knapsacksa.cpp:66
#15 0x000055555555534c in simulated_annealing () at knapsacksa.cpp:92
#16 0x0000555555555658 in main () at knapsacksa.cpp:146

46	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

Backtrace END
Memory content before bitflip:0x310xfb0x080x00
Fault Model: Random bit-flip
base address to flip value: 0x555555759034 
address max offset: 4
Memory content after  bitflip:0x540xa90x4b0x27
frame name: std::_Vector_base<int, std::allocator<int> >::~_Vector_base
symbol name: W2
symbol filename: knapsacksa.cpp
symbol line: 13
value: 659269972
value address: 0x555555759034 <W2>
Type: The type is an integer type.
Type sizeof: 4

[INFO -- 2018-12-08 14:16:55]
Fault Injection Successful after 3.793299674987793s
[INFO -- 2018-12-08 14:16:55]
event type: exit
[EXCEPTION -- 2018-12-08 14:16:55]
exit code: no exit code available
